### PR TITLE
Keyboard component revamp

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

Have pretty much copied the logic from KeyboardAvoidingView - https://github.com/facebook/react-native/blob/master/Libraries/Components/Keyboard/KeyboardAvoidingView.js

### CODE REVIEW

File: src/components/KeyboardCompatibleView/KeyboardCompatibleView.js [hunk 1]
```diff
@@ -20,63 +25,96 @@ import { KeyboardContext } from '../../context';
...
+            update: {
+              duration: e.duration > 10 ? e.duration : 10,
+              type: LayoutAnimation.Types[e.easing] || 'keyboard',
+            },
+          });
+        }
+        rootChannelView.current.measureInWindow((x, y) => {
+          if (Platform.OS === 'android') {
+            setChannelHeight(
+              isOpening
```
Comment (virdesai) [inline src/components/KeyboardCompatibleView/KeyboardCompatibleView.js:71]: based on the logic earlier, seems like `isOpening` will always be true for the android platform as it's only changed within an if statement that's iOS specific

File: src/components/KeyboardCompatibleView/KeyboardCompatibleView.js [hunk 2]
```diff
@@ -89,16 +127,12 @@ export const KeyboardCompatibleView = ({
...
-          toValue: height,
-          useNativeDriver: false,
-        }).start();
+      if (!initialHeight.current) {
+        initialHeight.current = height;
+        setChannelHeight(height);
       }
     },
-    [enabled, heightAnim, initialHeight],
+    [enabled],
```
Comment (virdesai) [inline src/components/KeyboardCompatibleView/KeyboardCompatibleView.js:135]: initialHeight not needed here anymore?

File: src/components/KeyboardCompatibleView/KeyboardCompatibleView.js [hunk 3]
```diff
@@ -20,63 +25,96 @@ import { KeyboardContext } from '../../context';
...
+  }, [enabled, appState]);
+
+  const dismissKeyboard = useCallback((callback) => {
+    if (!isKeyboardOpen.current) {
+      callback && callback();
+    } else {
+      if (callback) {
+        const subscribe = Keyboard.addListener('keyboardDidHide', () => {
+          callback();
+          subscribe.remove();
```
Comment (virdesai) [inline src/components/KeyboardCompatibleView/KeyboardCompatibleView.js:110]: so the subscribe const being set as this listener is being used within the callback of the same listener for the cleanup? does that work? seems a bit circular

File: src/components/KeyboardCompatibleView/KeyboardCompatibleView.js [hunk 4]
```diff
@@ -1,9 +1,14 @@
 import React, { useCallback, useEffect, useRef, useState } from 'react';
-import { Animated, Keyboard, View } from 'react-native';
-
-import { useKeyboardCompatibleHeight } from './hooks/useKeyboardCompatibleHeight';
+import {
+  Keyboard,
+  View,
+  LayoutAnimation,
+  Platform,
+  StatusBar,
```
[the remark this answers is not in the document]
Comment (vishalnarkhede) [reply src/components/KeyboardCompatibleView/KeyboardCompatibleView.js:7]: Damnn ... how come I keep forgetting about this one 😆

[the remark this answers is not in the document]
Comment (virdesai) [reply src/components/KeyboardCompatibleView/KeyboardCompatibleView.js:7]: lol soon enough it'll auto happen from the linting with release-2.0

**Pull request closed without merging**